src/opcode/mod.rs

```diff
@@ -106,7 +131,7 @@ impl PushValue {
         }
     }
 
-    pub fn eval(
+    pub fn eval_(
```

Comment:
Needs documentation. It will be confusing to have both `.eval()` and `.eval_()` in the public API on the same type, so also check if this can be private, and if so:
```suggestion
    fn eval_(
```
(or `pub(crate)` if necessary).